### PR TITLE
added change password feature

### PR DESCRIPTION
**Pull request by osadolortech**

### CODE REVIEW

File: tweetapi/login.py [hunk 1]
```diff
@@ -71,7 +71,14 @@ def get(self,request):
         }
         return response
 
-class ChangePassword(APIView):
-   pass
+class ChangePassword(CreateAPIView):
+    serializer_class = ChangeSerializers
+    def post(self, request, format=None):
+        serializer = ChangeSerializers(data=request.data,
```
Comment (iamr0b0tx) [inline tweetapi/login.py]: I already said you should look into `get_serializer_class` but you ignored it

File: tweetapi/registration.py [hunk 2]
```diff
@@ -36,21 +36,20 @@ class Meta:
...
+    password = serializers.CharField(write_only=True)
+    password_confirmation = serializers.CharField(write_only=True)
     class Meta:
-        model = User
         fields = (
-            'password','password_confirmation','new_password'
+            'password','password_confirmation'
         )
     def validate(self, attrs):
         user =self.context.get('user')
```
Comment (iamr0b0tx) [inline tweetapi/registration.py:47]: look into `require_context`

File: tweetapi/registration.py [hunk 3]
```diff
@@ -36,21 +36,20 @@ class Meta:
...
-        if attrs['password'] == attrs['new_password']:
-            raise serializers.ValidationError({'password': "new_password cant be same as old password"})
-        if attrs['new_password'] != attrs['password_confirmation']:
+        password = attrs.get('password')
+        password2= attrs.get('password_confirmation')
+        user = self.context.get('user')
+        if password != password2:
             raise serializers.ValidationError({"password":"password filed didnt match"})
```
Comment (iamr0b0tx) [inline tweetapi/registration.py]: spelling

**Review by iamr0b0tx — CHANGES_REQUESTED** [review]

**Comment by osadolortech** [comment]
Sorry sir I didn’t ignore, I actually did look into it just didn’t
implement it, will do that now sir..

File: tweetapi/registration.py (hunk 2 above)
Comment (osadolortech) [reply tweetapi/registration.py:47]: ok sir

File: tweetapi/registration.py (hunk 3 above)
Comment (osadolortech) [reply tweetapi/registration.py]: corrected

File: tweetapi/registration.py (hunk 2 above)
Comment (osadolortech) [reply tweetapi/registration.py:47]: sir so i checked for required_contex and no article had this well explained, not even on the DRF documentations, still looking for a better Materia for it.

Comment (iamr0b0tx) [reply tweetapi/registration.py:47]: I can't find it as well. I must have forgotten where I saw it

**Pull request merged**